addons/mail/models/mail_thread.py

```diff
@@ -1146,7 +1167,18 @@ def _message_route_process(self, message, message_dict, routes):
             else:
                 # if a new thread is created, parent is irrelevant
                 message_dict.pop('parent_id', None)
-                thread = ModelCtx.message_new(message_dict, custom_values)
+                # Report failure/record success of message creation except if alias is not defined (fallback model case)
+                try:
+                    thread = ModelCtx.message_new(message_dict, custom_values)
+                except:
+                    if alias:
+                        with self.pool.cursor() as cr:
+                            env = api.Environment(cr, self.env.user.id, {})
+                            env['mail.thread']._notify_alias_invalid(alias.id, message, message_dict)
+                        raise
```

Comment:
Yes indeed, if we do not execute that in another transaction, we cannot set the alias in error because the transaction will rollback so we need a new independent transaction.
It seems that “self.with_env(self.env(cr=new_cr))” is used in some place so I have used it as well, thanks.